src/index.js

```diff
@@ -93,15 +93,16 @@ async function run(request, ctx) {
   }
 
   // only use media handler when loaded via fstab. otherwise images are not processed.
-  let mediaHandler;
+  let mediaHandlerWithAuth;
+  let mediaHandlerWithoutAuth;
```

Comment:
this is a bit ugly... could enhance the media handler to allow to pass a function for the `auth` option, so this could be provided dynamically.

Reply:
added: https://github.com/adobe/helix-mediahandler/commit/b6c02a350e3cbe0a2409776483dbbd8aa4bea1c1#diff-583c6d206b4d52be0d0cfec9734827c41f39f3ece1de8dd4d232acb95fe3d03aR139

Reply:
love it, thank you.